src/connector/src/sink/mqtt.rs

```diff
@@ -127,6 +174,47 @@ impl Sink for MqttSink {
             )));
         }
 
+        if let Some(field) = &self.config.topic_field {
```

Comment:
Do you mind extracting this part of code to a separate function named like `get_topic_field_index_path` and writing some unit tests for it?

Reply:
+1 seems too hack here.